--- a/src/graph/index.rs
+++ b/src/graph/index.rs
@@ -511,7 +511,7 @@
                     // The task assigned to each round of `set_element`.
                     let future = async move {
                         self_clone
-                            .set_chunk(&context_clone, &batch_clone, &ids_clone, r)
+                            .set_chunk(&context_clone, batch_clone.as_ref(), &ids_clone, r)
                             .await
                     };
 
